File: crowdfunding-backend/src/main/java/com/example/project/service/interfaces/MessageService.java
package com.example.project.service.interfaces;

import com.example.project.dto.MessageRequestDTO;
import com.example.project.dto.MessageResponseDTO;

import java.util.List;

public interface MessageService {
    
    MessageResponseDTO sendMessage(MessageRequestDTO request);
    
    List<MessageResponseDTO> getConversation(Long user1Id, Long user2Id);
    
    List<MessageResponseDTO> getRecentConversations(Long userId);
    
    void markAsRead(Long messageId, Long userId);
    
    long getUnreadCount(Long userId);
}

File: crowdfunding-frontend/src/services/messageService.js
import api from './api';

const messageService = {
  // Envoyer un message
  sendMessage: async (messageData) => {
    const response = await api.post('/messages', messageData);
    return response.data;
  },

  // Récupérer une conversation entre deux utilisateurs
  getConversation: async (user1Id, user2Id) => {
    const response = await api.get(`/messages/conversation`, {
      params: { user1Id, user2Id }
    });
    return response.data;
  },

  // Récupérer les conversations récentes d'un utilisateur
  getRecentConversations: async (userId) => {
    const response = await api.get(`/messages/recent/${userId}`);
    return response.data;
  },

  // Marquer un message comme lu
  markAsRead: async (messageId, userId) => {
    const response = await api.patch(`/messages/${messageId}/read`, null, {
      params: { userId }
    });
    return response.data;
  },

  // Obtenir le nombre de messages non lus
  getUnreadCount: async (userId) => {
    const response = await api.get(`/messages/unread/${userId}`);
    return response.data;
  }
};

export default messageService;
